### ISSUE TITLE
Runkit - ES6 Template Literals do not work on dev.to

### ISSUE THREAD

**Issue by shardyMBAI** [forem/forem#1431]
**Describe the bug**
When writing the markup for a dev.to post using `runkit` for code examples template literals (backticks `) are stripped out.

**To Reproduce**

1. Go to dev.to
2. Click on write a post
3. add
```js
const test = 'test everything everywhere'
const str = `this is a ${test}`
console.log(str)
```
4. Click on preview to  see error (image attched)
![_error-1](https://user-images.githubusercontent.com/27963753/50564392-8258fe00-0cf2-11e9-9a20-af757585dd65.png)

**Expected behavior**
template literals used in dev.to's runkit embedded environment to be supported
Here is an example from [runkits demo page](https://npm.runkit.com/demo)
<img width="549" alt="screen shot 2018-12-31 at 11 46 33 am" src="https://user-images.githubusercontent.com/27963753/50564467-10cd7f80-0cf3-11e9-90a1-5c3a91ae7c0c.png">

**Screenshots**
Markdown:
![_error-2](https://user-images.githubusercontent.com/27963753/50564475-204cc880-0cf3-11e9-8adc-364ce16b4d6c.png)

Result: 
![_error-1](https://user-images.githubusercontent.com/27963753/50564478-25117c80-0cf3-11e9-89b2-183c6f57c1e8.png)

Expected:
<img width="549" alt="screen shot 2018-12-31 at 11 46 33 am" src="https://user-images.githubusercontent.com/27963753/50564481-2b9ff400-0cf3-11e9-81d5-dd83db5e4037.png">

- OS: MAC
- Browser: Chrome
- Version: Version 71.0.3578.98 (Official Build) (64-bit)

**Additional context**
The bug was discovered while updating a post as a request PR here: https://github.com/shardy613/es6-notes/pull/13

**Comment by maestromac** [comment]
Thank you for reporting this issue. This is most likely caused by this method https://github.com/thepracticaldev/dev.to/blob/master/app/labor/markdown_parser.rb#L130

**Comment by shardy613** [comment]
That looks like the culprit!

Thank you for taking time to look and label the issue.

**Comment by Zhao-Andy** [comment]
Still an issue unfortunately.

**Comment by jessleenyc** [comment]
@mariocsee is this resolved? saw there's a merged PR referenced.

**Comment by shardyMBAI** [comment]
Hey @jessleenyc,

I just tested again with this content:

<img width="899" alt="Screen Shot 2019-07-31 at 10 16 03 PM" src="https://user-images.githubusercontent.com/27963753/62260699-dfc84100-b3e0-11e9-9e29-12df6dd9cba1.png">

and using the preview feature I received this result:

<img width="910" alt="Screen Shot 2019-07-31 at 10 15 55 PM" src="https://user-images.githubusercontent.com/27963753/62260718-f40c3e00-b3e0-11e9-8c3b-fb4b60d5bfe2.png">


Unfortunately I believe this is still an issue.

Cheers.

**Comment by rianadon** [comment]
It looks like this is an issue of the markdown parser running on the runkit code.

If I use HTML code inside the runkit tag like:
```javascript
console.log('This content is <i>Italic</i>');
```
it renders as such:

<img width="298" src="https://user-images.githubusercontent.com/18223213/67617990-deb76d80-f79d-11e9-8a7e-44624463ee4f.png" />

The `<i>` tags are stripped out.

**Comment by rianadon** [comment]
Adding a code fence around the JavaScript but inside the RunKit tag fixes the issue.

    {% runkit %}
    ```
    const test = 'test everything everywhere'
    const str = `this is a ${test}`
    console.log(str)
    ```
    {% endrunkit %}

It would seem like the easiest fix would be to introduce a preprocessing step in the markdown renderer. We would find all runkit tags and edit their bodies to contain code blocks around their contents before running the Markdown parser.

However, this approach would fail the following scenario:

    `{% runkit %}` is a tag!

Since the Liquid tag loses its meaning once wrapped within a code block.

The only bug-free method of solving the issue I can think of would be to use two passes over the markdown. The first renders it as-is and keeps track of which liquid tags are parsed and which are left as-is. The second uses this information to replace the parsed liquid tags of type runkit with code fences.

This solution sounds like vastly engineering a problem that can be solved by documentation, so maybe we could mention in the `runkit` docs that surrounding the code in code blocks solves parsing errors and is recommended.

**Comment by wolfhoundjesse** [comment]
Would this solely require adding an explanation to *views/pages/_editor_liquid_help.html.erb* and *views/pages/_editor_guide_text.html.erb*? E.g.:

> If your code contains Template Literals, you can fence the code block to preserve the backticks and prevent a parsing error:
````
{% runkit %}
```  
console.log(`Good morning, ${user}`)
```
{% endrunkit %}
````

I'm feeling this issue because I just researched how to put backticks in code blocks in Markdown on GitHub. 😂

**Comment by cmgorton** [comment]
Looks like this might be a duplicate of this issue https://github.com/forem/forem/issues/1228 . I am going to close this issue and anymore discussion on a possible fix for this can be added to #1228
